apps/storefront/src/pages/QuoteDraft/index.test.tsx

```diff
@@ -3854,4 +3855,263 @@ describe('when the user is a B2B customer', () => {
+      isDefaultShipping: 0,
+      isDefaultBilling: 0,
+    };
+
+    const defaultShippingAddress = buildAddressWith({
+      country: country.countryCode,
+      state: state.stateName,
+    });
+
+    const defaultBillingAddress = buildAddressWith({
+      country: country.countryCode,
+      state: state.stateName,
+    });
+
+    const companyInfo = buildCompanyStateWith({
+      companyInfo: { status: CompanyStatus.APPROVED },
+      customer: {
+        userType: UserTypes.MULTIPLE_B2C,
+        role: CustomerRole.SENIOR_BUYER,
+        emailAddress: customerEmail,
+      },
+      permissions: [
+        {
+          code: 'create_quote',
+          permissionLevel: 2,
+        },
+      ],
+    });
+
+    const product = buildDraftQuoteItemWith({
+      node: {
+        primaryImage: 'url',
+        quantity: 1,
+        variantSku: 'test',
+        basePrice: 10,
+        taxPrice: 5,
+        productName: 'Test Product',
+        productsSearch: buildProductWith({
+          inventoryLevel: 10,
+          inventoryTracking: 'product',
+          sku: 'test',
+          basePrice: '10.00',
+          offeredPrice: '10.00',
+          productId: 1,
+          imageUrl: 'url',
+          id: 4451490883947128,
+        }),
+      },
+    });
+
+    const quoteInfo = buildQuoteInfoStateWith({
+      draftQuoteInfo: {
+        contactInfo: { email: customerEmail },
+      },
+      draftQuoteList: [product],
+    });
+
+    const getPreloadedState = (
+      billingAddress: Address = noAddress,
+      shippingAddress: Address = noAddress,
+    ) => ({
+      preloadedState: {
+        company: companyInfo,
+        storeInfo: storeInfoWithDateFormat,
+        quoteInfo: {
+          ...quoteInfo,
+          draftQuoteInfo: {
+            ...quoteInfo.draftQuoteInfo,
+            billingAddress,
+            shippingAddress,
+          },
+        },
+      },
+    });
+
+    const createQuoteMutation = vi.fn();
+
+    beforeEach(() => {
+      vi.clearAllMocks();
+      set(window, 'b2b.callbacks.dispatchEvent', vi.fn().mockReturnValue(true));
+      server.use(
+        graphql.query('Countries', () => HttpResponse.json({ data: { countries: [country] } })),
+        graphql.query('Addresses', () =>
+          HttpResponse.json({
+            data: {
+              addresses: {
+                totalCount: 2,
+                edges: [{ node: { ...billingAddress } }, { node: { ...shippingAddress } }],
+              },
+            },
+          }),
+        ),
+        graphql.query('getQuoteExtraFields', () =>
+          HttpResponse.json({ data: { quoteExtraFieldsConfig: [] } }),
+        ),
+        graphql.query('GetQuoteInfoB2B', () =>
+          HttpResponse.json({ data: { quote: { id: '272989', quoteNumber: '911911' } } }),
+        ),
+        http.post('*/api/v2/extra-fields/quote/validate', () => HttpResponse.json({ code: 200 })),
+        graphql.mutation('CreateQuote', async ({ request }: { request: Request }) => {
```

Comment:
🤔 I don't think you need these types:


```suggestion
        graphql.mutation('CreateQuote', async ({ request }) => {
```

Reply:
we are fetching query out of request and if i remove the type then i am getting eslint issue.
I think it is good to type

Reply:
you can use `query` to fetch the query from the request. This is mistyped:

```tsx
graphql.mutation('CreateQuote', async ({ query }) => {
  createQuoteMutation(query);
```